### PR TITLE
PRs and CPU runners for cutlass

### PR DESCRIPTION
**Pull request by RaulPPelaez**
Cutlass takes about 24 hours to compile in the current runners. I am trying to make it run on the openstack ones.

## Guidelines for marking packages as broken:

* We prefer to patch the repo data (see [here](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock))
  instead of marking packages as broken. This alternative workflow makes environments more reproducible.
* Packages with requirements/metadata that are too strict but otherwise work are
  not technically broken and should not be marked as such.
* Packages with missing metadata can be marked as broken on a temporary basis
  but should be patched in the repo data and be marked unbroken later.
* In some cases where the number of users of a package is small or it is used by
  the maintainers only, we can allow packages to be marked broken more liberally.
* We (`conda-forge/core`) try to make a decision on these requests within 24 hours.

What will happen when a package is marked broken?

* Our bots will add the `broken` label to the package. The `main` label will remain on the package and this is normal.
* Our bots will rebuild our repodata patches to remove this package from the repodata.
* In a few hours after the `anaconda.org` CDN picks up the new patches, you will no longer be able to install the package from the `main` channel.

## Checklist:

* [ ] I want to mark a package as broken (or not broken):
  * [ ] Added a description of the problem with the package in the PR description.
  * [ ] Pinged the team for the package for their input.

* [ ] I want to archive a feedstock:
  * [ ] Pinged the team for that feedstock for their input.
  * [ ] Make sure you have opened an issue on the feedstock explaining why it was archived.
  * [ ] Linked that issue in this PR description.
  * [ ] Added links to any other relevant issues/PRs in the PR description.

* [x] I want to request (or revoke) access to an opt-in CI resource:
  * [ ] Pinged the relevant feedstock team(s)
  * [x] Added a small description explaining why access is needed

### CODE REVIEW

File: requests/cutlass.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+action: cirun
+feedstocks:
+  - cutlass
+resources:
+  - cirun-openstack-gpu-large
+  - cirun-openstack-cpu-large
+pull_request: true
+revoke: false
```
Comment (h-vetinari) [inline requests/cutlass.yaml:8]:
```suggestion
revoke: false
send_pr: true
```

File: requests/cutlass.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+action: cirun
+feedstocks:
+  - cutlass
+resources:
+  - cirun-openstack-gpu-large
+  - cirun-openstack-cpu-large
```
Comment (h-vetinari) [inline requests/cutlass.yaml:6]: I don't actually think we'll need the CPU version, as cutlass only has GPU builds.

**Review by h-vetinari — APPROVED** [review]
So this is basically a copy of #881 but for cutlass, which is a good start; however, pytorch was the very first, and there were a few more iterations since then.

It's better to take the [example](https://github.com/conda-forge/admin-requests/blob/main/examples/example-open-gpu-server.yml) as a reference.

File: requests/cutlass.yaml (hunk 2 above)
Comment (RaulPPelaez) [reply requests/cutlass.yaml:6]: The GPU I do not think its actually needed. We are going to compile but not test.
We could test on the V100 available in the self-hosted runner, but I believe it is a waste to hold the GPU during 24h just for that...

Comment (RaulPPelaez) [reply requests/cutlass.yaml:6]: In that sense cutlass would benefit from ci_4xlarge, which has 8 cores and a lot of RAM

Comment (h-vetinari) [reply requests/cutlass.yaml]: > We could test on the V100 available in the self-hosted runner, but I believe it is a waste to hold the GPU during 24h just for that...

The openstack runner is _much_ more powerful than the generic azure agents. I think if the pytorch megabuild runs through with `cirun-openstack-gpu-large`, it's unlikely we'll need something bigger for cutlass. But on that I'll defer to @jaimergp and @isuruf

Comment (jaimergp) [reply requests/cutlass.yaml]: Let's try with `large` first and if it proves to be insufficient, we can check xlarge. Agree that there's no GPU required in this case.

File: requests/cutlass.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+action: cirun
+feedstocks:
+  - cutlass
+resources:
+  - cirun-openstack-cpu-4xlarge
```
Comment (jaimergp) [inline requests/cutlass.yaml:5]:
```suggestion
  - cirun-openstack-cpu-large
```

**Comment by RaulPPelaez** [comment]
Thanks guys, always a pleasure.

**Pull request merged**